misc/mdns/src/dns.rs

```diff
@@ -211,6 +206,40 @@ fn append_u16(out: &mut Vec<u8>, value: u16) {
     out.push((value & 0xff) as u8);
 }
 
+/// If a peer ID is longer than 63 characters, split it into segments to
+/// be compatible with RFC 1035.
+fn segment_peer_id(peer_id: String) -> String {
+    // Guard for the most common case
+    if peer_id.len() <= MAX_LABEL_LENGTH { return peer_id }
+
+    // This will only perform one allocation except in extreme circumstances.
+    let mut out = String::with_capacity(peer_id.len() + 8);
+
+    for (idx, chr) in peer_id.chars().enumerate() {
```

Comment:
This line bothers me a bit. We know the string only contains ASCII characters, so I guess it's fine. I would have preferred using bytes.